File: strip_sub.py
#!/bin/python
import sys, os, hashlib, time
from pathlib import Path

def attemptFix():
    
    fileName = sys.argv[1]
    
    time.sleep(5)
    
    for i in range(5):
       if(os.path.exists(fileName) == True):
           break
       else:
           time.sleep(5)
    
    if(os.path.exists(fileName) == False):
        return   
        
    with open(fileName, "r") as slnFile:
        lines = slnFile.readlines()
        
    slnFile.close()
        
    with open(fileName, "w") as outFile:
        skipCount = 0
        for line in lines:
            if skipCount == 0:
                if line.find("ALL_BUILD") == -1:
                    outFile.write(line)
                else:
                    skipCount = 4
            else:
                skipCount -= 1
    outFile.close()
    return        

FILE_LOCK_NAME = '.cmake_config_fix.lock'

try:
    with open(FILE_LOCK_NAME, 'r') as f:
        print('Script is locked (Probably already running)')
        exit()

except IOError:
    with open(FILE_LOCK_NAME, 'w') as f:
        f.write('')
        f.close()

attemptFix()

os.remove(FILE_LOCK_NAME)

File: strip_sub.lua
function file_exists(fileName)

    local file=io.open(fileName, "r")

    if file~=nil then 

        file:close()
        return true

    else

        return false

    end

end

-- Actual Code

function attemptFix()

    fileName = arg[1]

    --[[ for i = 1, 5 do

        if file_exists(fileName) then
            break
        else
            sleep(3)
        end

    end--]]

    if file_exists(fileName) == false then
        return
    end

    local lines = {}

    for line in io.lines(fileName) do
        table.insert(lines, line)
    end

    slnFile = io.open(fileName, "w")

    for i, l in ipairs(lines) do
        if string.find(l, "ALL_BUILD") == nil then
            slnFile:write(l .. "\n")
        end
    end

    slnFile:close()

end

--if file_exists(".cmake_config_fix.lock") then
    --print("Script is locked (Probably already running)")
    --return
--end

--local FILE_LOCK_NAME = '.cmake_config_fix.lock'
--local lockFile = io.open(FILE_LOCK_NAME, "w")

--lockFile:write(" ")
--lockFile:close()

attemptFix()
--os.remove(FILE_LOCK_NAME)